lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h

```diff
@@ -0,0 +1,1001 @@
+   */
+  uint32_t member_present_mask; /* bitmask, see significance of bits below */
+  uint32_t member_valid_mask;   /* bitmask, see significance of bits below */
+  uint8_t m5gsregistrationtype;
+  amf_procedures_t* amf_procedures;
+  uint num_registration_request; /* Num registration request received */
+
+  // imsi present mean we know it but was not checked with identity proc, or was
+  // not provided in initial message
+  imsi_t _imsi;     /* The IMSI provided by the UE or the AMF, set valid when
+                       identification returns IMSI */
+  imsi64_t _imsi64; /* The IMSI provided by the UE or the AMF, set valid when
+                       identification returns IMSI */
+  imsi64_t saved_imsi64; /* Useful for 5.4.2.7.c */
+  imei_t _imei;          /* The IMEI provided by the UE                     */
+  imeisv_t _imeisv;      /* The IMEISV provided by the UE                   */
+  guti_m5_t _m5_guti;    /* The GUTI assigned to the UE                     */
+  guti_m5_t m5_old_guti; /* The GUTI assigned to the UE                     */
+  ksi_t ksi;             /*key set identifier  */
+  drx_parameter_t _drx_parameter;
+  int remaining_vectors;  // remaining unused vectors
+  m5g_auth_vector_t
+      _vector[MAX_EPS_AUTH_VECTORS]; /* 5GMM authentication vector */
+  amf_security_context_t
+      _security; /* Current 5GMM security context: The security context which
+                    has been activated most recently. Note that a current 5GMM
+                                       //security context originating from
+                    either a mapped or native 5GMM security context may exist
+                    simultaneously with a native
+                                      // non-current 5GMM security context.*/
+
+  int amf_cause; /* AMF failure cause code                          */
+  amf_fsm_state_t amf_fsm_state;
+  void* t3422_arg;
+  smf_context_t smf_context;              // smf contents
+  drx_parameter_t _current_drx_parameter; /* stored TAU Request IE Requirement
+                                             AMF24.501R15_5.5.3.2.4_4*/
+
+  // TODO: DO BETTER  WITH BELOW
+  std::string smf_msg; /* SMF message contained within the initial request*/
+  bool is_imsi_only_detach;
+};
+
+class amf_ue_context_t {
+ public:
+  amf_ue_context_t() {}
+  ~amf_ue_context_t() {}
+
+  /* hash_table_uint64_ts_t is defined in lib/hastable*/
+  hash_table_uint64_ts_t* imsi_amf_ue_id_htbl;    // data is amf_ue_ngap_id_t
+  hash_table_uint64_ts_t* tun11_ue_context_htbl;  // data is amf_ue_ngap_id_t
+  hash_table_uint64_ts_t*
+      gnb_ue_ngap_id_ue_context_htbl;             // data is amf_ue_ngap_id_t
+  obj_hash_table_uint64_t* guti_ue_context_htbl;  // data is amf_ue_ngap_id_t
+};
+
+enum m5gmm_state_t {
+  UE_UNREGISTERED = 0,
+  UE_REGISTERED,
+};
+
+enum m5gcm_state_t {
+  M5GCM_IDLE = 0,
+  M5GCM_CONNECTED,
+};
+
+class ue_m5gmm_context_s {
```

Comment:
again defining this as class does not make much sense if it has only public variables and no class methods.

Reply:
changed to struct